openhands/sdk/conversation/conversation.py

```diff
@@ -108,8 +120,54 @@ def run(self) -> None:
             # and check will we be able to execute .send_message
             # BEFORE the .run loop finishes?
             with self.state:
+                # clear the flag before calling agent.step() (user approved)
+                if self.state.agent_waiting_for_confirmation:
+                    self.state.agent_waiting_for_confirmation = False
+
                 # step must mutate the SAME state object
                 self.agent.step(self.state, on_event=self._on_event)
+
+            # In confirmation mode, stop after one iteration if waiting for confirmation
+            if self.state.agent_waiting_for_confirmation:
+                break
+
             iteration += 1
             if iteration >= self.max_iteration_per_run:
                 break
+
+    def get_pending_actions(self) -> list[ActionEvent]:
+        """Get all actions that are waiting for confirmation from the agent."""
+        return self.agent.get_pending_actions(self.state)
```

Comment:
Actually, since all this doing is just calling `get_unmatched_actions(state.events)`, can we completly remove this `get_pending_actions` method, and directly call `get_unmatched_actions(state.events)` here